C7Engine/EntryPoints/UnitInteractions.cs

```diff
@@ -1,98 +1,98 @@
 namespace C7Engine
 {
-    using C7GameData;
-    using System;
-    using System.Collections.Generic;
+	using C7GameData;
+	using System;
+	using System.Collections.Generic;
 
-    public class UnitInteractions
-    {
+	public class UnitInteractions
+	{
 
-        private static Queue<MapUnit> waitQueue = new Queue<MapUnit>();
+		private static Queue<MapUnit> waitQueue = new Queue<MapUnit>();
 
-        public static MapUnit getNextSelectedUnit(GameData gameData)
-        {
-            foreach (Player player in gameData.players) {
-                //TODO: Should pass in a player GUID instead of checking for human
-                //This current limits us to one human player, although it's better
-                //than the old limit of one non-barbarian player.
-                if (player.isHuman) {
-                    foreach (MapUnit unit in player.units) {
-                        if (unit.movementPointsRemaining > 0 && !unit.isFortified) {
-                            if (!waitQueue.Contains(unit)) {
-                                return unit;
-                            }
-                        }
-                    }
-                }
-            }
-            if (waitQueue.Count > 0) {
-                return waitQueue.Dequeue();
-            }
-            return MapUnit.NONE;
-        }
+		public static MapUnit getNextSelectedUnit(GameData gameData)
+		{
+			foreach (Player player in gameData.players) {
+				//TODO: Should pass in a player GUID instead of checking for human
+				//This current limits us to one human player, although it's better
+				//than the old limit of one non-barbarian player.
+				if (player.isHuman) {
+					foreach (MapUnit unit in player.units) {
+						if (unit.movementPointsRemaining > 0 && !unit.IsBusy()) {
+							if (!waitQueue.Contains(unit)) {
+								return UnitWithAvailableActions(unit);
```

Comment:
Any particular reason we're switching this back?  Flintlock changed it last week at https://github.com/C7-Game/Prototype/pull/169/files#diff-3309f3270d6b4945af6fac63b05a6230c15e3167db67577293ed3dbfe72f1f84R22, so that Game.getSelectedUnit grabs the actions, fixing a bug described on that PR.  I don't think it necessarily harms anything to add it here, but I don't know what the benefit would be.

Reply:
Nope that's a bad rebase on my part 😦 I'll double check the diff to make sure nothing else is changed